tests/lax_scipy_test.py

```diff
@@ -190,6 +190,22 @@ def testLogSumExpNans(self):
         result = lsp_special.logsumexp(1.0, b=1.0)
         self.assertEqual(result, 1.0)
 
+  @jtu.sample_product(
+    shape=[(1,), (2,), (3,), (4,), (5,)],
+    dtype=float_dtypes,
+  )
+  def testLogSumExpWhere(self, shape, dtype):
+    rng = jtu.rand_default(self.rng())
+    x = rng(shape, dtype)
+    rng = jtu.rand_bool(self.rng())
+    mask = rng(shape, bool)
+    rng = jtu.rand_int(self.rng(), high=shape[0])
+    i = rng(shape, int)
+    mask[i] = True  # guarantee at least one True element
```

Comment:
It doesn't seem bad to test the case with no True elements

Reply:
`scipy.special.logsumexp` raises a ValueError for an empty array:

```sh
$ python3 -c "import scipy; print(scipy.special.logsumexp([]))"
ValueError: zero-size array to reduction operation maximum which has no identity
```

This error comes from the use of `numpy.max` inside scipy's implementation:

```sh
$ python3 -c "import numpy; print(numpy.max([]))"
ValueError: zero-size array to reduction operation maximum which has no identity
```

But scipy forgot to set `initial=-inf` for it.

Mathematically, the correct answer for the empty array is log(empty sum) = log(0) = -inf.

Reply:
Then something like this would work:
```python
y_expected = osp_special.logsumexp(x[mask]) if np.any(mask) else jnp.array(-np.inf, dtype=dtype)
```

Reply:
I like that! Added.